spark/src/main/scala/eval/directive/RDDOpDirectives.scala

```diff
@@ -1,118 +1,309 @@
+      case (RDDResult(r) :: IntResult(i) :: Nil) => RDDResult(ri(r, i))
+      case (IntResult(i) :: RDDResult(r) :: Nil) => RDDResult(ri(r, i))
+      case (RDDResult(r) :: DoubleResult(d) :: Nil) => RDDResult(rd(r, d))
+      case (DoubleResult(d) :: RDDResult(r) :: Nil) => RDDResult(rd(r, d))
+    }
+
+    Valid(results)
+  }
+
+  val pow = Directive { case (Pow(_), res1 :: res2 :: Nil) =>
+    val f: (RDD[(SpatialKey, Tile)], RDD[(SpatialKey, Tile)]) => RDD[(SpatialKey, Tile)] = { (a,b) =>
+      a.combineValues(b) { (t1,t2) => t1.combineDouble(t2)(math.pow) }
+    }
+
+    Valid(reduce({_ localPow _}, { (i,r) => r.localPow(i) }, { _ localPow _ }, { (d,r) => r.localPow(d) }, f, res1, res2))
+  }
+
+  /* --- UNARY EXPRESSIONS --- */
+
+  val classify = Directive { case (Classification(_, classMap), RDDResult(r) :: Nil) =>
+    unary({ _.color(ColorMap(classMap.classifications)) }, r)
+  }
+
+  val sin = Directive { case (Sin(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.sin(_))}, { i => DoubleResult(math.sin(i.toDouble)) }, { d => DoubleResult(math.sin(d)) }, res)
+  }
+  val cos = Directive { case (Cos(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.cos(_))}, { i => DoubleResult(math.cos(i.toDouble)) }, { d => DoubleResult(math.cos(d)) }, res)
+  }
+  val tan = Directive { case (Tan(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.tan(_))}, { i => DoubleResult(math.tan(i.toDouble)) }, { d => DoubleResult(math.tan(d)) }, res)
+  }
+
+  val sinh = Directive { case (Sinh(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.sinh(_))}, { i => DoubleResult(math.sinh(i.toDouble)) }, { d => DoubleResult(math.sinh(d)) }, res)
+  }
+  val cosh = Directive { case (Cosh(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.cosh(_))}, { i => DoubleResult(math.cosh(i.toDouble)) }, { d => DoubleResult(math.cosh(d)) }, res)
+  }
+  val tanh = Directive { case (Tanh(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.tanh(_))}, { i => DoubleResult(math.tanh(i.toDouble)) }, { d => DoubleResult(math.tanh(d)) }, res)
+  }
+
+  val asin = Directive { case (Asin(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.asin(_))}, { i => DoubleResult(math.asin(i.toDouble)) }, { d => DoubleResult(math.asin(d)) }, res)
+  }
+  val acos = Directive { case (Acos(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.acos(_))}, { i => DoubleResult(math.acos(i.toDouble)) }, { d => DoubleResult(math.acos(d)) }, res)
+  }
+  val atan = Directive { case (Atan(_), res :: Nil) =>
+    mathy({_.localMapDouble(math.atan(_))}, { i => DoubleResult(math.atan(i.toDouble)) }, { d => DoubleResult(math.atan(d)) }, res)
+  }
+
+  val floor = Directive { case (Floor(_), res :: Nil) =>
+    mathy({ _.localFloor }, { IntResult(_) }, { d => IntResult(math.floor(d).toInt) }, res)
+  }
+  val ceil = Directive { case (Ceil(_), res :: Nil) =>
+    mathy({ _.localCeil }, { IntResult(_) }, { d => IntResult(math.ceil(d).toInt) }, res)
+  }
```

Comment:
`floor`, `ceil`, and `round` are made to return `IntResult` here, which follows intuition but not Scala. In Scala, `floor: Double => Double`.